medio/backends/itk_io.py

```diff
@@ -43,13 +44,28 @@ def read_img(
         :return: numpy image and metadata object which includes pixdim, affine, original orientation string and
         coordinates system
         """
+        if header:
+            imageio = itk.GDCMImageIO.New()
+            if private_tags:
+                imageio.LoadPrivateTagsOn()
+            # we need to set the imageio to the reader, but with fallback_only=True it will not set it unless it fails
+            fallback_only = False
+        else:
+            imageio = None
         input_path = Path(input_path)
         if input_path.is_dir():
             img = ItkIO.read_dir(
-                str(input_path), pixel_type, fallback_only, series=series, header=header
+                str(input_path), pixel_type, fallback_only, series, imageio
             )
         elif input_path.is_file():
-            img = ItkIO.read_img_file(str(input_path), pixel_type, fallback_only)
+            # If the input is not a dicom, fallback to not use imageio.
+            # NOTE: this assume imageio is used for dicom files only. If it changed, we need to modify this.
+            is_dicom = imageio.CanReadFile(str(input_path))
```

Comment:
`imageio` can be `None` here (and also below) - try for example:
```python
arr, meta = medio.read_img("example.nii.gz", backend="itk")
```
Or even
```python
arr, meta = medio.read_img("dicom-series/", backend="itk", header=False)
```

Reply:
something happened during the merge from master. thanks for catching it!